nixos/modules/services/network-filesystems/samba.nix

```diff
@@ -191,6 +176,100 @@ in
         networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ 139 445 ];
         networking.firewall.allowedUDPPorts = mkIf cfg.openFirewall [ 137 138 ];
       })
-    ];
 
+      (lib.mkIf cfg.nmbd.enable {
```

Comment:
I think now got a lot of duplication. Maybe we could work with a common attrset instead?

Reply:
From my point of view, I'm not sure it's worth it.

There was a common attr set before, however the configuration values of the 3 services is quite different between each.

That's why I split the configuration of each service in this PR.